cel/io_test.go

```diff
@@ -18,11 +18,172 @@ import (
 	"testing"
 
 	"github.com/google/cel-go/checker/decls"
+	"github.com/google/cel-go/common/types"
+	"github.com/google/cel-go/common/types/ref"
+	"github.com/google/cel-go/common/types/traits"
+	"github.com/google/cel-go/test/proto3pb"
 	"google.golang.org/protobuf/proto"
 
 	exprpb "google.golang.org/genproto/googleapis/api/expr/v1alpha1"
 )
 
+func TestValueToRefValue(t *testing.T) {
+	stdEnv, err := NewEnv()
+	if err != nil {
+		t.Fatalf("NewEnv() failed: %v", err)
+	}
+
+	var expected ref.Val
+	var val *exprpb.Value
+	var actual ref.Val
```

Comment:
For these tests, I would recommend setting up a parameterized test:

```go
tests := []struct{
  value interface{}
} {
  {value: types.NullValue},
  {value: []interface{}{true, "abc"}},
  ...
}

for i, tst := range tests {
  tc := tst
  t.Run(fmt.Printf("[%d]%v", i, tc.value), func(t *testing.T) {
    refVal := env.TypeAdapter().NativeToValue(tc.value)
    val, err := RefValueToValue(refVal)
    if err != nil {
       t.Fatalf("RefValueToValue(%v) failed with error: %v", refVal, err)
    }
    actual, err := ValueToRefValue(env.TypeAdapter(), val)
    if err != nil { ... }
    if refVal.Equal(actual) != types.True {
       t.Errorf("got val %v, wanted %v", actual, refVal)
    }
  })
}

Reply:
It should. The `NativeToValue` call will make sure the `value` provided is a `ref.Val` and from there you can do the round-trip and compare whether the original `ref.Val` equals the round-tripped `ref.Val`